sdks/java/core/src/main/java/org/apache/beam/sdk/io/fs/ResourceId.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.fs;
+
+import org.apache.beam.sdk.io.fs.ResolveOptions.StandardResolveOptions;
+
+/**
+ * An identifier which represents a resource.
+ *
+ * <p>{@code ResourceId} is hierarchical and composed of a sequence of directory
+ * and file name elements separated by a special separator or delimiter.
+ */
+public interface ResourceId {
+
+  /**
+   * Resolves the given {@code String} against this {@code ResourceId}.
+   *
+   * <p>In order to write file system agnostic code, callers should not include delimiters
+   * in {@code other}, and should use {@link StandardResolveOptions} to specify
+   * whether to resolve a file or a directory.
+   *
+   * <p>For example:
+   *
+   * <pre>{@code
+   * ResourceId homeDir = ...
```

Comment:
what is in the `...` here. How is a caller expected to ever get an instance of `ResourceId`?